### ISSUE TITLE
Feature request: IPv6 google_compute_address

### ISSUE THREAD

**Issue by JeanDaniel-Fischer** [hashicorp/terraform-provider-google#5027]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Add support for IP Version in the google compute address resource so we can reserved IPv6 instead of IPv4 when do not need a v4. This will be very usefull to stop using v4.

The idea is to match the --ip-version of [gcloud](https://cloud.google.com/sdk/gcloud/reference/compute/addresses/create)

### New or Affected Resource(s)

* google_compute_address

### Potential Terraform Configuration

```tf
resource "google_compute_address" "static_ip" {
  name = "static-ip"
  project = "my-project"
  
  ip_version = "IPV6"
}
```

**Comment by Chupaka** [comment]
Isn't that about "IP version can only be specified for global addresses that are generated automatically" and 
resource "google_compute_global_address" "default_v6" {
  name       = "global-ip-v6"
  ip_version = "IPV6"
}
?

**Comment by Unichron** [comment]
I think the confusion comes from the documentation mess on google's side. They have two resources:

- https://cloud.google.com/compute/docs/reference/rest/v1/addresses
- https://cloud.google.com/compute/docs/reference/rest/v1/globalAddresses

The intro texts on both of these are the same and have very inconsistent settings, for example in theory the 'address' has an ipVersion field but you are only supposed to use it for global addresses (WTF).

**Comment by sanghaniJ** [comment]
b/262667864

**Comment by sanghaniJ** [comment]
- The request is for the resource `google_compute_address` where support is required for field `ipVersion` which is possible via [gCloud](https://cloud.google.com/sdk/gcloud/reference/compute/addresses/create) cli.
 
- The field `ip_version` is already supported in `google_compute_global_address`.

- The requested field can be found in [API Docs](https://cloud.google.com/compute/docs/reference/rest/beta/addresses) for Addresses. But, addresses have a common API structure in case of  both global and regional.

- In addition to that, we can find the description for `ipVersion` stating `This can only be specified for a global address.` So, it seems the API/Resource Behavior doesn't support the request. 

Hence, we can close this issue.

**Comment by sanghaniJ** [comment]
Additionally, I tried adding the field `ip_version` to `google_compute_address` to verify if the API behavior is similar to the documented one. Here's the config which I passed in tf. 

```
resource "google_compute_address" "ip_address" {
  name = "my-address"
  ip_version = "IPV6"
}
```

And after terraform apply, I received the expected error i.e `Error: Error creating Address: googleapi: Error 400: IP Version is not supported for regional addresses., badRequest` which verified the behavior.
